pybind/include/flatbuffers/pybind/memory.h

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2024 Figure AI, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FLATBUFFERS_PYBIND_MEMORY_H_
+#define FLATBUFFERS_PYBIND_MEMORY_H_
+
+#include <pybind11/pybind11.h>
+
+#include "flatbuffers/allocator.h"
+#include "flatbuffers/flatbuffer_builder.h"
+#include "flatbuffers/stl_emulation.h"
+
+namespace flatbuffers {
+namespace pybind {
+
+/// @brief Custom allocator for python which wraps a resizable bytearray.
+/// This allocator is expected to have a shorter lifetime than the bytearray.
+class ByteArrayAllocator : public Allocator {
+ public:
+  explicit ByteArrayAllocator(pybind11::bytearray bytearray)
```

Comment:
I think it makes more sense to take this by reference then copy it into the member variable than to take this by copy then move it into the member variable. That also allows you to avoid the copy in CreateFlatBufferBuilder as that can just take a reference too.